packages/ra-ui-materialui/src/button/IconButtonWithTooltip.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { MouseEvent } from 'react';
+import { IconButton, IconButtonProps, Tooltip } from '@material-ui/core';
+import { useTranslate } from 'ra-core';
+
+/**
+ * An IconButton with a tooltip which ensures the tooltip is closed on click to avoid ghost tooltips
+ * when the button position changes.
+ */
+export const IconButtonWithTooltip = ({
+    label,
+    onClick,
+    ...props
+}: IconButtonWithTooltipProps) => {
+    const translate = useTranslate();
+    const [open, setOpen] = React.useState(false);
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    const handleOpen = () => {
+        setOpen(true);
+    };
+
+    const translatedLabel = translate(label, { _: label });
+
+    const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+        handleClose();
+        onClick(event);
+    };
+
+    return (
+        <Tooltip
+            title={translatedLabel}
+            open={open}
```

Comment:
I don't understand why you have to handle the open state manually. Tooltip does it on its own, see e.g. https://github.com/marmelab/react-admin/blob/ce116c1059500b8b8758281968f2916bac7b116e/packages/ra-ui-materialui/src/layout/UserMenu.tsx#L86-L95

Reply:
It's explained in the component main comment